pkgs/development/libraries/wayland/scanner.nix

```diff
@@ -0,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  wayland,
+  meson,
+  pkg-config,
+  ninja,
+  wayland-scanner,
+  expat,
+  libxml2,
+}:
+
+stdenv.mkDerivation {
+  pname = "wayland-scanner";
+  inherit (wayland) version src;
+
+  outputs = [
+    "out"
+    "bin"
+    "dev"
+  ];
+  separateDebugInfo = true;
+
+  mesonFlags = [
+    "-Ddocumentation=false"
+    "-Dlibraries=false"
+    "-Dtests=false"
+  ];
+
+  depsBuildBuild = [ pkg-config ];
```

Comment:
I see this was in the original `wayland` derivation but without a comment why do we need `BuildBuild` in addition to `BuildHost` 🤔

Reply:
This is required for any package that uses pkg-config to find a build dependency.  pkg-config always needs to be one step "up" from the packages it's finding.

Reply:
(For this reason, if it's non-obvious and isn't already, it would make a lot more sense for this to be explained in doc/, rather than in every package that builds programs for the build system that use a library)

Reply:
> This is required for any package that uses pkg-config to find a build dependency

Oh, actually, this makes sense. Thanks

Reply:
Hmmh, but this would be a general cross-compilation docs change, unrelated to this wayland PR, no?

Reply:
> Hmmh, but this would be a general cross-compilation docs change, unrelated to this wayland PR, no?

Yes, out of scope